python/rmm/_cuda/stream.pyx

```diff
@@ -117,7 +119,10 @@ cdef class Stream:
         self._cuda_stream, self._owner = stream._cuda_stream, stream._owner
 
 
-DEFAULT_STREAM = Stream._from_cudaStream_t(cuda_stream_default.value())
+if int(os.environ.get("RMM_PER_THREAD_DEFAULT_STREAM", "0")) != 0:
+    DEFAULT_STREAM = Stream._from_cudaStream_t(cuda_stream_per_thread.value())
+else:
+    DEFAULT_STREAM = Stream._from_cudaStream_t(cuda_stream_default.value())
```

Comment:
Do we want to allow this to be changed during runtime of an application or only at import time?

Reply:
I would assume at import time as this is what I think cupy does now

Reply:
Well CuPy reads the env var (same is done here) so technically there it could be changed at runtime by overwriting `os.environ['CUPY_PER_THREAD_DEFAULT_STREAM']`. However such operation is really a bad practice that is hard to be fool-proofed.